src/lib.rs

```diff
@@ -24,6 +28,101 @@ where
     K: AsRef<str>,
 {
     api_key: K,
+    client: reqwest::Client,
+}
+
+#[non_exhaustive]
```

Comment:
Two things about this

1. I'm not sure it needs to be `#[non_exhaustive]`. I think non-exhaustive is primarily for structs we expect the consumer to match on, and we want to be able to add variants without breaking their code. This type will primarily be constructed by the consumer, not matched on (I think) so it's probably not necessary. Doesn't hurt anything though, I learned today by looking at [this](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) that the non-exhaustive attribute has no effect in the defining crate, so we still get the warnings about non-exhaustive patterns (which is good!)
2. The lower case variants look a little weird to my eye, I think they could be capitalized normally, since we have to turn them into strings anyway, so we can format them however we like here, but I don't mind that much

Reply:
My idea was to have them be as similar to the actual mimetype names as possible, but I agree it does come at the cost of being less Rust-like. The same goes for the language codes below.

Reply:
Yep makes sense, totally your call, I don't mind very much any which way!